kernel/kernel-api/src/main/java/io/delta/kernel/internal/snapshot/LogSegment.java

```diff
@@ -235,6 +236,9 @@ public LogSegment(
             () ->
                 Stream.concat(checkpoints.stream(), deltas.stream()).collect(Collectors.toList()));
 
+    // Make sure all delta log files are valid.
+    assertLogFilesBelongToTable(logPath, allFiles.get());
```

Comment:
move this to above

    ////////////////////////////////
    // Member variable assignment //
    ////////////////////////////////